react/Search.tsx

```diff
@@ -24,45 +27,66 @@ const Search: FC<Props> = ({ query }) => {
+
+  if (results.length < 1 && isMobile) return <SearchBar />
 
   return (
     <Fragment>
-      <h1 className="t-heading-1 center">
-        {results.length > 0
-          ? `Results for "${queryString}"`
-          : 'No Results Found'}
-      </h1>
-      <ul className="list pl0 center mt5 mw8">
-        {results
-          ? results.map((result: SearchResult, index: number) => (
-              <li
-                key={index}
-                className={`mh8 mv8 searchResult ${
-                  isNotLastResult(results, index) ? 'searchResultBorder' : ''
-                }`}>
-                <h2 className="t-heading-2 searchTitle">{result.title}</h2>
-                <p className="t-body searchSnippet">{result.snippet}</p>
-                <a
-                  className={`link c-emphasis no-underline t-body ml-auto
+      {isMobile ? <SearchBar /> : ''}
+      {!loading ? (
+        <div className={`pv6 w-100 flex flex-column`}>
+          <h1
+            className={`t-heading-1 right ph6 ${isMobile ? '' : 'center'}`}
+            style={{ fontSize: `${isMobile ? '24px' : '48px'}` }}>
+            {results.length > 0
+              ? `Results for "${queryString}"`
+              : 'No Results Found'}
+          </h1>
+          <ul className="w-100 list pl0 center mw8">
+            {results
+              ? results.map((result: SearchResult, index: number) => (
+                  <li
+                    key={index}
+                    className={`mh8 ${
+                      isFirstResult(index) ? '' : 'mv8'
+                    } searchResult ${
+                      isNotLastResult(results, index)
+                        ? 'searchResultBorder'
+                        : ''
+                    }`}>
+                    <h2
+                      className="t-heading-2 searchTitle"
+                      style={{ wordBreak: 'break-word' }}>
+                      {result.title}
+                    </h2>
+                    <p className="t-body searchSnippet">{result.snippet}</p>
+                    <a
+                      className={`link c-emphasis no-underline t-body ml-auto
                 flex items-center dim`}
-                  href={result.link}>
-                  <div className="flex flex-column flex-row-l">
-                    <span className="dn db-l">Read More</span>
-                    <div className="ml5">
-                      <RightArrow />
-                    </div>
-                  </div>
-                </a>
-              </li>
-            ))
-          : null}
-      </ul>
+                      href={result.link}>
+                      <div className="flex flex-column flex-row di">
+                        <span>Read More</span>
+                        <div className="ml5">
+                          <RightArrow />
+                        </div>
+                      </div>
+                    </a>
+                  </li>
+                ))
+              : null}
```

Comment:
The conditional for this block of code could also be `results && results.map()...`, which would make this a bit cleaner since you don't need to add `null` down in the bottom, also it makes it easier to read :)

Reply:
Shouldn't this `conditional` wrap the `<ul>` tag as well?
In the case `results` is falsy, this code would render an empty `<ul>` into the DOM

Reply:
Ah, I see, thanks!